aws/kops-aws-platform/iam-authenticator.tf

```diff
@@ -115,19 +129,21 @@ data "aws_iam_policy_document" "admin" {
 }
 
 resource "aws_iam_role" "admin" {
+  count              = "${var.kops_iam_enabled == "true" ? 1 : 0}"
   name               = "${module.kops_admin_label.id}"
   assume_role_policy = "${data.aws_iam_policy_document.admin.json}"
   description        = "The Kops admin role for aws-iam-authenticator"
 }
 
 module "iam_authenticator_config" {
-  source                = "git::https://github.com/cloudposse/terraform-aws-kops-iam-authenticator-config.git?ref=tags/0.1.1"
+  enabled               = "${var.kops_iam_enabled}"
+  source                = "git::https://github.com/cloudposse/terraform-aws-kops-iam-authenticator-config.git?ref=tags/0.2.0"
   cluster_id            = "${var.cluster_id}"
   kube_config_path      = "${var.kube_config_path}"
-  admin_iam_role_arn    = "${aws_iam_role.admin.arn}"
+  admin_iam_role_arn    = "${element(concat(aws_iam_role.admin.*.arn, list("")), 0)}"
```

Comment:
```suggestion
  admin_iam_role_arn    = "${join("", aws_iam_role.admin.*.arn)}"
```

Reply:
This is the [canonical idiom from the Terraform docs](https://www.terraform.io/upgrade-guides/0-11.html#referencing-attributes-from-resources-with-count-0), and in public code I like to follow conventions when it is reasonable. I agree that `join` is shorter and cleaner, but is there a functional difference?

Reply:
o functional difference (we just use it everywhere in our code)